### DISCUSSION TITLE
Support for Open Graph protocol

### DISCUSSION THREAD

**Discussion by aapnetin** [outline/outline#3532] [General] [answered]
Tested on twitter https://cards-dev.twitter.com/validator for Twitter card support, it looks Outlines don't support .

am i missing something or there is no support for Open Graph protocol ?

**Comment by tommoor** [comment, marked as answer]
We can/should add it – however it will only work for publicly shared links

**Reply by jtviolet** [reply]
Would a PR be accepted to add deeper Open Graph integrations?

**Reply by tommoor** [reply]
What kind of functionality are you thinking of? If it's more of this kind of metadata on publicly shared links then that seems totally fine

**Reply by jtviolet** [reply]
In particular, when URLs are shared in a system (e.g. Slack), it unfurls with:

![Screenshot 2025-02-21 at 10 39 21 AM](https://github.com/user-attachments/assets/be096f8b-36dd-48f6-b32c-b584e21514f7)

Ideally, it would utilize the workspace name for the title, something from the document (or if not, a default site description) for the meta descripion, logo or uploaded image in settings for the og:image, etc.

**Reply by tommoor** [reply]
We do already support Slack unfurling of private links, not sure if you set this up?

https://docs.getoutline.com/s/hosting/doc/slack-G2mc8DOJHk#h-link-unfurling

**Comment by jtviolet** [comment]
Ah! What about Microsoft Teams? I tested there as well.


Thanks,

*Jordan Violet*
P:216.903.5351 | L:linkedin.com/in/jtviolet | A:Ohio, USA 45844
Vice President, Community & Engagement @ Netwrix

**Reply by tommoor** [reply]
Well it's not so much about OpenGraph in these case – because the information is behind a login you have to build a full integration where we can determine if the user has the appropriate permissions to view the document content. So, this won't be possible until that #2093 is created

**Comment by jtviolet** [comment]
That all makes sense. I’d like to explore it. In addition, would you be
opposed to adding the ability for it to pull from your organization name
and (a newly added) description field?


Thanks,

*Jordan Violet*
P:216.903.5351 | L:linkedin.com/in/jtviolet | A:Ohio, USA 45844
Vice President, Community & Engagement @ Netwrix

**Reply by tommoor** [reply]
I would suggest creating an issue for what you want to achieve and we can discuss/agree there, but it sounds... fine?

**Comment by jtviolet** [comment]
Awesome! I’d love to.


Thanks,

*Jordan Violet*
P:216.903.5351 | L:linkedin.com/in/jtviolet | A:Ohio, USA 45844
Vice President, Community & Engagement @ Netwrix
